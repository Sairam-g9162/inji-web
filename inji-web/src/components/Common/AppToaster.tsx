import React from "react";
import { ToastContainer, toast } from "react-toastify";
import { useSelector } from "react-redux";
import { RootState } from "../../types/redux";
import { isRTL } from "../../utils/i18n";

const CustomToast = ({ message }: { message: string }) => (
    <div data-testid="custom-toast">{message}</div>
);

export const AppToaster: React.FC = () => {
    const language = useSelector((state: RootState) => state.common.language);
    const positionClass = isRTL(language) ? "top-left" : "top-right";

    // Example of triggering a toast
    React.useEffect(() => {
        toast(<CustomToast message="Test message" />);
    }, []);

    return (
        <div className={`toast-container ${positionClass}`} data-testid="toast-container">
            <ToastContainer
                position={isRTL(language) ? "top-left" : "top-right"}
                autoClose={1000}
                hideProgressBar
                newestOnTop
                closeOnClick
                rtl={isRTL(language)}
                icon={<React.Fragment />}
                pauseOnFocusLoss
                draggable
                pauseOnHover
                style={{ width: '400px' }}
                theme="colored"
            />
        </div>
    );
};
