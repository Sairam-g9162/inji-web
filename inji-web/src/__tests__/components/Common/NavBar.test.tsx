import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import { BrowserRouter, Routes, Route } from 'react-router-dom';
import { NavBar } from '../../../components/Common/NavBar';
import { NavBarProps } from '../../../types/components';

// Mock the SearchCredential component
jest.mock('../../../components/Credentials/SearchCredential', () => ({
    SearchCredential: () => <div data-testid="SearchCredential">Search</div>,
}));

const defaultProps: NavBarProps = {
    title: 'Test Title',
    link: '/test-link',
    search: true,
};

const renderWithProvider = (ui: React.ReactElement, { route = '/' } = {}) => {
    window.history.pushState({}, 'Test page', route);
    return render(
        <BrowserRouter>
            <Routes>
                <Route path="*" element={ui} />
                <Route path="/test-link" element={<div>Test Link Page</div>} />
            </Routes>
        </BrowserRouter>
    );
};

describe("NavBar Component Layout Tests", () => {
    test('check the presence of the navbar container', () => {
        const{asFragment} = renderWithProvider(<NavBar {...defaultProps} />);
        expect(asFragment()).toMatchSnapshot();
    });
});

describe("NavBar Component Functionality Tests", () => {
    test('check navigates to the link when back arrow is clicked', () => {
        renderWithProvider(<NavBar {...defaultProps} />);
        fireEvent.click(screen.getByTestId('NavBar-Back-Arrow'));
        expect(screen.getByText('Test Link Page')).toBeInTheDocument();
    });
});
