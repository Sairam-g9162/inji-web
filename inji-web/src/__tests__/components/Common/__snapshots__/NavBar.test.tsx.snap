// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar Component Layout Tests check the presence of the navbar container 1`] = `
<DocumentFragment>
  <div
    class="bg-iw-navigationBar text-iw-title p-4 py-10"
    data-testid="NavBar-Outer-Container"
  >
    <nav
      class=" mx-auto flex flex-col justify-start container items-start
 sm:flex-row sm:justify-start sm:items-center"
      data-testid="NavBar-Inner-Container"
    >
      <div
        class="flex items-center my-5 sm:my-0"
      >
        <div
          class="cursor-pointer"
        >
          <svg
            data-testid="NavBar-Back-Arrow"
            fill="currentColor"
            height="24"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 512 512"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M244 400 100 256l144-144M120 256h292"
              fill="none"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="48"
            />
          </svg>
        </div>
        <span
          class="text-2xl font-semibold ps-2 whitespace-nowrap"
          data-testid="NavBar-Text"
        >
          Test Title
        </span>
      </div>
      <div
        data-testid="SearchCredential"
      >
        Search
      </div>
    </nav>
  </div>
</DocumentFragment>
`;
